src/components/DiffView/index.tsx

```diff
@@ -253,30 +195,19 @@ export class DiffViewBase extends React.Component<Props> {
         })}
       </div>
     );
-  }
-}
+  };
 
-const mapStateToProps = (
-  state: ApplicationState,
-  ownProps: PublicProps,
-): PropsFromState => {
-  const { version } = ownProps;
-
-  let linterMessages;
-  const map = selectMessageMap(state.linter, version.id);
-  if (map) {
-    linterMessages = map[version.selectedPath]
-      ? map[version.selectedPath]
-      : // No messages exist for this path.
-        null;
-  }
+  render() {
+    const { version } = this.props;
 
-  return {
-    linterMessages,
-    linterMessagesAreLoading: state.linter.isLoading,
-  };
-};
+    return (
+      <LinterProvider version={version}>
+        {this.renderWithMessages}
+      </LinterProvider>
+    );
+  }
+}
 
-export default withRouter<PublicProps & Partial<DefaultProps> & RouterProps>(
-  connect(mapStateToProps)(DiffViewBase),
-) as React.ComponentType<PublicProps & Partial<DefaultProps>>;
+export default withRouter(DiffViewBase) as React.ComponentType<
+  PublicProps & Partial<DefaultProps>
+>;
```

Comment:
Is this still needed here? Can't we type the `withRouter` HOC instead?

```
withRouter<PublicProps & Partial<DefaultProps> & RouterProps>(DiffViewBase);
```

I believe if `PublicProps` and `DefaultProps` were the same type, we would not even have to add types to `withRouter` (it works in `CodeView`):

```
withRouter(DiffViewBase);
```